packages/mocks/src/handlers.ts

```diff
@@ -48,7 +49,12 @@ const upsertTransaction = graphql.mutation(
       return response(context.data({ upsertTransaction: newTransaction }));
     }
 
-    return response(context.data({}));
+    const idx = TransactionData.findIndex(
+      ({ id: filterId }) => id === filterId
+    );
+    TransactionData[idx] = { ...TransactionData[idx], ...patch };
+
+    return response(context.data({ upsertTransaction: TransactionData[idx] }));
```

Comment:
This handler was just handling new upserts. Took the rest of the code I forgot to add from the mock server to also handle actual upserts